python/src/utils.rs

```diff
@@ -244,3 +245,45 @@ impl Hnsw {
         self.vectors.to_data().to_pyarrow(py)
     }
 }
+
+/// A newtype wrapper for a Lance type.
+///
+/// This is used for types that have a corresponding dataclass in Python.
+pub struct PyLance<T>(pub T);
+
+impl<T> IntoPy<PyObject> for PyLance<T>
+where
+    Self: ToPyObject,
+{
+    fn into_py(self, py: Python) -> PyObject {
+        self.to_object(py)
+    }
+}
+
+pub fn extract_vec<'a, T>(ob: &Bound<'a, PyAny>) -> PyResult<Vec<T>>
+where
+    PyLance<T>: FromPyObject<'a>,
+{
+    ob.extract::<Vec<PyLance<T>>>()
+        .map(|v| v.into_iter().map(|t| t.0).collect())
+}
+
+pub fn export_vec<'a, T>(py: Python<'a>, vec: &'a [T]) -> Vec<PyObject>
```

Comment:
Can we document these methods?  `extract_vec` I guess is pretty obvious since `extract` is well-used in pyo3 but `export_vec` is maybe a touch confusing.